qt/FilterBar.cc

```diff
@@ -232,9 +232,16 @@ FilterBar::FilterBar(Prefs& prefs, TorrentModel const& torrents, TorrentFilter c
     connect(&myTorrents, SIGNAL(modelReset()), this, SLOT(recountSoon()));
     connect(&myTorrents, SIGNAL(rowsInserted(QModelIndex, int, int)), this, SLOT(recountSoon()));
     connect(&myTorrents, SIGNAL(rowsRemoved(QModelIndex, int, int)), this, SLOT(recountSoon()));
-    connect(&myTorrents, SIGNAL(dataChanged(QModelIndex, QModelIndex)), this, SLOT(recountSoon()));
     connect(myRecountTimer, SIGNAL(timeout()), this, SLOT(recount()));
 
+    connect(&myTorrents, &TorrentModel::torrentsChanged, [this](torrent_ids_t const& /*ids*/, Torrent::fields_t const& fields)
```

Comment:
In light of recent fixes in #1107, please pass `this` as context to trigger signal disconnects on object destruction.